tools/mtouch/BuildTasks.mtouch.cs

```diff
@@ -120,8 +120,13 @@ class RunRegistrarTask : BuildTask {
 
 		protected override void Execute ()
 		{
-			Target.StaticRegistrar.Generate (Target.Assemblies.Select ((a) => a.AssemblyDefinition), RegistrarHeaderPath, RegistrarCodePath, out var initialization_name, Target.App.ClassMapPath);
+			var assemblies = Target.Assemblies.Select ((a) => a.AssemblyDefinition);
+			Target.StaticRegistrar.Generate (assemblies, RegistrarHeaderPath, RegistrarCodePath, out var initialization_name, out var typeMap);
 			RegistrationMethods.Add (initialization_name);
+			if (Target.App.OptimizeClassHandles) {
+				var rewriter = new Rewriter (assemblies, typeMap);
+				rewriter.Process ();
+			}
 		}
```

Comment:
This chunk of code is duplicated in a few places, would it make sense to move it inside the StaticRegistrar.Generate method so that there's only one copy of it?

Reply:
That's a good idea - I didn't do it initially because I was worried about exception handling (which this code needs).